scene/3d/visual_instance_3d.cpp

```diff
@@ -150,43 +150,89 @@ Ref<Material> GeometryInstance3D::get_material_override() const {
 }
 
-float GeometryInstance3D::get_lod_max_distance() const {
-	return lod_max_distance;
+float GeometryInstance3D::get_visibility_range_end() const {
+	return visibility_range_end;
 }
 
-void GeometryInstance3D::set_lod_min_hysteresis(float p_dist) {
-	lod_min_hysteresis = p_dist;
-	RS::get_singleton()->instance_geometry_set_draw_range(get_instance(), lod_min_distance, lod_max_distance, lod_min_hysteresis, lod_max_hysteresis);
+void GeometryInstance3D::set_visibility_range_begin_hysteresis(float p_dist) {
+	visibility_range_begin_hysteresis = p_dist;
+	RS::get_singleton()->instance_geometry_set_visibility_range(get_instance(), visibility_range_begin, visibility_range_end, visibility_range_begin_hysteresis, visibility_range_end_hysteresis);
 }
 
-float GeometryInstance3D::get_lod_min_hysteresis() const {
-	return lod_min_hysteresis;
+float GeometryInstance3D::get_visibility_range_begin_hysteresis() const {
+	return visibility_range_begin_hysteresis;
 }
 
-void GeometryInstance3D::set_lod_max_hysteresis(float p_dist) {
-	lod_max_hysteresis = p_dist;
-	RS::get_singleton()->instance_geometry_set_draw_range(get_instance(), lod_min_distance, lod_max_distance, lod_min_hysteresis, lod_max_hysteresis);
+void GeometryInstance3D::set_visibility_range_end_hysteresis(float p_dist) {
+	visibility_range_end_hysteresis = p_dist;
+	RS::get_singleton()->instance_geometry_set_visibility_range(get_instance(), visibility_range_begin, visibility_range_end, visibility_range_begin_hysteresis, visibility_range_end_hysteresis);
 }
 
-float GeometryInstance3D::get_lod_max_hysteresis() const {
-	return lod_max_hysteresis;
+float GeometryInstance3D::get_visibility_range_end_hysteresis() const {
+	return visibility_range_end_hysteresis;
+}
+
+void GeometryInstance3D::set_visibility_range_parent(const Node *p_parent) {
+	const GeometryInstance3D *gi = Object::cast_to<GeometryInstance3D>(p_parent);
+
+	if (!gi) {
+		return;
+	}
+
+	RS::get_singleton()->instance_geometry_set_visibility_parent(get_instance(), gi->get_instance());
+}
+
+void GeometryInstance3D::clear_visibility_range_parent() {
+	RS::get_singleton()->instance_geometry_set_visibility_parent(get_instance(), RID());
+}
+
+void GeometryInstance3D::_find_visibility_range_children(Node *p_node) const {
+	GeometryInstance3D *gi = Object::cast_to<GeometryInstance3D>(p_node);
+	if (gi && gi != this) {
+		gi->set_visibility_range_parent(this);
+	}
+
+	for (int i = 0; i < p_node->get_child_count(); i++) {
+		Node *child = p_node->get_child(i);
+		if (!child->get_owner()) {
+			continue; //maybe a helper
+		}
+
+		_find_visibility_range_children(child);
+	}
+}
+
+void GeometryInstance3D::_attach_visibility_range_children() const {
+	for (int i = 0; i < visibility_range_children.size(); i++) {
+		Node *n = get_node_or_null(visibility_range_children[i]);
+		if (n) {
+			_find_visibility_range_children(n);
+		}
+	}
 }
 
 void GeometryInstance3D::_notification(int p_what) {
+	if (p_what == NOTIFICATION_ENTER_TREE) {
```

Comment:
should probably use NOTIFICATION_READY to make sure children are properly configured (and call request_ready() afterwards so its called again next time it enters the tree.)

Reply:
I don't think this is needed anymore. Right now, the only requirement is for the node to be inside a tree. If any nodes are added after that, they will automatically inherit the correct visibility parent.